src/renderer/dx/CustomTextLayout.cpp

```diff
@@ -63,6 +63,19 @@ CustomTextLayout::CustomTextLayout(gsl::not_null<IDWriteFactory1*> const factory
 
         _text += text;
     }
+
+    const auto textLength = gsl::narrow<UINT32>(_text.size());
+
+    BOOL isTextSimple = FALSE;
+    UINT32 uiLengthRead = 0;
+    UINT32 glyphStart = 0;
+
+    HRESULT hr = S_OK;
+
+    _glyphIndices.resize(textLength);
+
+    hr = _analyzer->GetTextComplexity(_text.c_str(), textLength, _font.Get(), &isTextSimple, &uiLengthRead, &_glyphIndices.at(glyphStart));

```

Comment:
I'm not sure how to handle the HR faliure here.

Reply:
Maybe we'd want to do a `SUCCEEDED(hr) && isTextSimple && uiLengthRead == textLength` in the line below?